In [2]:
import pandas as pd
import numpy as np


df = pd.read_csv('csv/ab_nyc_2019.csv')

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [3]:
df['last_review'] = pd.to_datetime(df['last_review'], format ='%Y-%m-%d')

# 1.Qual o `id` do imóvel com o aluguel (diária) mais caro da base de dados?

Selecionar a coluna price e ordenar de forma decrescente/ Apresentar o resultado da primeira linha

In [4]:
mais_cara = df[df['price'] == df['price'].max()].iloc[0, 0]
mais_cara

7003697

# 2. Qual o `id` do imóvel com o aluguel (diária) mais barato da base de dados?

Selecionar o minimo do valor da coluna price

In [5]:
mais_barato = df[df['price'] == df['price'].min()].iloc[0, 0]
mais_barato

18750597

1620248

# 3. Qual o `id` do imóvel que foi mais locado da base de dados?

Selecionar o valor máximo da coluna 'calculated_host_listings_count'

In [6]:
mais_locado = df[df['calculated_host_listings_count'] == df['calculated_host_listings_count'].max()].iloc[0, 0]
mais_locado

30181691

9145202

# 4. Qual o `id` do imóvel que ficou mais tempo com o anúncio disponível, em dias, para locação na base de dados?

Selecionar o valor máximo da coluna 'availability_365'

In [7]:
mais_tempovago = df[df['availability_365'] == df['availability_365'].max()].iloc[0, 0]
mais_tempovago

2539

# 5. Qual o `id` do imóvel que ficou menos tempo com o anúncio disponível, em dias, para locação na base de dados?

Selecionar o valor mínimo da coluna 'availability_365

In [8]:
menos_tempovago = df[df['availability_365'] == df['availability_365'].min()].iloc[0, 0]
menos_tempovago

5022

# 6. O imóvel com o maior valor de aluguel (diária) da base de dados é o imóvel que possui mais avaliações na base de dados?

Buscar o número máximo de avaliações em um imovel
verificar se o imovel com maior valor de aluguel apresenta esse valor

In [9]:

maximo_aval = df['reviews_per_month'].max()
maximo_aval

mais_cara = df[df['price'] == df['price'].max()].iloc[0, 13]
mais_cara

print(f'A casa mais cara possui {mais_cara}, já a casa que possui um maior numero de avaliaçãoes apresenta {maximo_aval}, desa forma a resposta é: NÃO')

A casa mais cara possui 0.04, já a casa que possui um maior numero de avaliaçãoes apresenta 58.5, desa forma a resposta é: NÃO


# 7. O imóvel que possui a menor quantidade mínima de diárias para locação é
também o imóvel que possui o aluguel mais caro?

Buscar o imovel que possui menor quantidade de diarias 'minimum_nights' e relacionar com o 
imovel com maior valor de aluguel


In [10]:

menor_diaria = df['minimum_nights'].min()
menor_diaria

mais_cara = df[df['price'] == df['price'].max()].iloc[0, 10]
mais_cara

print(f'A casa mais cara possui {mais_cara} diárias, já a casa que possui um menor numero apresenta {menor_diaria} diária, dessa forma a resposta é: NÃO')

A casa mais cara possui 100 diárias, já a casa que possui um menor numero apresenta 1 diária, dessa forma a resposta é: NÃO


# 8. Qual é a média do número mínimo de diárias para locação de um imóvel?

Selecionar a coluna 'minimum_nights' aplicar uma operação matematica para saber o valor médio

In [11]:
media_y = df['minimum_nights'].mean()
media_y

7.029962163820431

# 9. Qual é o `id` do imóvel com a menor quantidade mínima de diárias para locação da base de dados?

Selecionar o menor valor da coluna 'minimum_nights' e localizar o id desse imovel




In [12]:
id_menor_diaria = df[df['minimum_nights'] == df['minimum_nights'].min()].iloc[0, 0]
id_menor_diaria


2539

# 10. Qual é o `id` do host que possui o imóvel mais alugado na base de dados?

Selecionar o valor máximo da coluna 'calculated_host_listings_count' e localizar o seu id

In [13]:
id_maior_locacao = df[df['calculated_host_listings_count'] == df['calculated_host_listings_count'].max()].iloc[0, 2]
id_maior_locacao

219517861

47621202

# 11. Qual é o `id` do host que possui o imóvel menos alugado na base de dados?

repetir o mesmo que o anterior, porém selecionando o mínimo

In [14]:
id_menor_locacao = df[df['calculated_host_listings_count'] == df['calculated_host_listings_count'].min()].iloc[0, 2]
id_menor_locacao

4632

c = 82685

# 12. Qual é o `id` do host que possui o imóvel com mais avaliações na base de dados?

In [15]:

df1 = df[df['reviews_per_month']> 1]
df1

id_mais_avaliac = df1[df1['reviews_per_month'] == df1['reviews_per_month'].max()].iloc[0, 2]
id_mais_avaliac

244361589

47621202

# 13. Qual é o `id` do host que possui a maior quantidade de imóveis cadastrados na base de dados?

Verificar a quantidade de vezes que cada 'id do host' aparece nas linhas
Através do agrupamendo do id do host, somar a quantidade de imoveis



In [16]:
imoveis_por_id = df[['id','host_id']].groupby('host_id').sum().sort_values('id', ascending = False).reset_index().iloc[0, 0]
imoveis_por_id

219517861

# 14. Qual o `id` do host que possui o imóvel com a última avaliação feita na base de dados?

Ordenar a coluna 'last_review' e selecionar o id do host

In [17]:


id_host_last_review = df[['host_id','last_review']].sort_values('last_review', ascending = False).reset_index().iloc[0,1]
id_host_last_review

74162901

526653

# 15. Qual o `id` do host que possui mais imóveis ativos dentro da base de dados?

In [19]:

id_host_count = df7[['host_id','reviews_per_month']].sort_values('reviews_per_month', ascending = False).reset_index()
id_host_count

index    host_id  reviews_per_month
0      42075  244361589              58.50
1      42076  244361589              27.95
2      38870  228415932              20.94
3      27287  156684502              19.75
4      28651  156684502              17.82
...      ...        ...                ...
48879  48890    8232441                NaN
48880  48891    6570630                NaN
48881  48892   23492952                NaN
48882  48893   30985759                NaN
48883  48894   68119814                NaN

[48884 rows x 3 columns]

In [ ]:
219517861

# 16. O host que possui mais imóveis é o host que também possui mais avaliações?

mais avaliações 244361589
mais imoveis 219517861

In [130]:
#Sim

# 17. Qual é a categoria que mais possui imóveis dentro da base de dados?

In [133]:
categoria_mais = df[['id','room_type']].groupby('room_type').sum().sort_values('id', ascending = False).reset_index().iloc[0, 0]
categoria_mais

'Entire home/apt'

# 18. Qual é a categoria que menos possui imóveis dentro da base de dados?

In [134]:
categoria_menos = df[['id','room_type']].groupby('room_type').sum().sort_values('id', ascending = True).reset_index().iloc[0, 0]
categoria_menos


'Shared room'